drivers/ethernet/eth_lm3s6965.c

```diff
@@ -0,0 +1,313 @@
+ /* LM3S6965 Ethernet Controller
+ *
+ * Copyright (c) 2018 Zilogic Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "dev/eth_lm3s6965"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+
+#include <logging/sys_log.h>
+#include <net/ethernet.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <device.h>
+#include <soc.h>
+#include <ethernet/eth_stats.h>
+#include "eth_lm3s6965_priv.h"
+
+static void eth_lm3s6965_assign_mac(struct device *dev)
+{
+	u32_t value = 0x0;
+
+	value |= CONFIG_ETH_LM3S6965_MAC0;
+	value |= CONFIG_ETH_LM3S6965_MAC1 << 8;
+	value |= CONFIG_ETH_LM3S6965_MAC2 << 16;
+	value |= CONFIG_ETH_LM3S6965_MAC3 << 24;
+	sys_write32(value, REG_MACIA0);
+
+	value = 0x0;
+	value |= CONFIG_ETH_LM3S6965_MAC4;
+	value |= CONFIG_ETH_LM3S6965_MAC5 << 8;
+	sys_write32(value, REG_MACIA1);
+}
+
+static void eth_lm3s6965_dev_init(struct device *dev)
+{
+	u32_t value;
+
+	ARG_UNUSED(dev);
+
+	/* 1. Assign MAC address to Hardware */
+	eth_lm3s6965_assign_mac(dev);
+
+	/* 2. Program MCRCTL to clear RXFIFO */
+	value = BIT_MACRCTL_RSTFIFO;
+	sys_write32(value, REG_MACRCTL);
+
+	/* 3. Enable transmitter */
+	value = BIT_MACTCTL_DUPLEX | BIT_MACTCTL_CRC |
+		BIT_MACTCTL_PADEN | BIT_MACTCTL_TXEN;
+	sys_write32(value, REG_MACTCTL);
+
+	/* 4. Enable Receiver */
+	value = BIT_MACRCTL_BADCRC | BIT_MACRCTL_RXEN;
+	sys_write32(value, REG_MACRCTL);
+}
+
+static int eth_lm3s6965_send(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct net_eth_hdr *pkt_hdr;
+	struct device *dev;
+	struct eth_lm3s6965_runtime *dev_data;
+	u32_t reg_val;
+	int bytes_left;
+	u8_t *data_buf;
+
+	dev = net_if_get_device(iface);
+	dev_data = DEV_DATA(dev);
+
+	if (!pkt->frags) {
+		SYS_LOG_ERR("No data to send");
+		return -ENODATA;
+	}
+
+	/* transmit the frame*/
+	/* First two bytes are data_len for frame */
+	bytes_left = net_pkt_get_len(pkt) + net_pkt_ll_reserve(pkt);
+	data_buf = net_pkt_ll(pkt);
+	reg_val = net_pkt_get_len(pkt);
+	reg_val |= ((u32_t)(*data_buf++) << 16);
+	reg_val |= ((u32_t)(*data_buf++) << 24);
+
+	/* Send the first word */
+	sys_write32(reg_val, REG_MACDATA);
+	bytes_left -= 2;
+
+	/* Write the rest of data in net_buf */
+	for (bytes_left; bytes_left > 3; bytes_left -= 4) {
```

Comment:
What about the remaining fragments?

Reply:
fixed the bug, pertaining to remaining fragments.